From c13f43124fae0d36bb4f6e56627d63e2576d97ce Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Thu, 21 Oct 2010 11:45:50 +0800
Subject: [PATCH 188/505] ENGR00132842 MX50 pxp unit test: correct the update
 parameters

1. upd_data.use_alt_buffer is not initialized, then the epdc driver
thinks alter buffer is used. This is not correct, so fixed.
2. use virtual x instead of actual x for fb according to edpc driver.

Signed-off-by: Robby Cai <R63905@freescale.com>
---
 test/pxp_lib_test/pxp_test.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/pxp_lib_test/pxp_test.c b/test/pxp_lib_test/pxp_test.c
index 6762d66..a287d1c 100644
--- a/test/pxp_lib_test/pxp_test.c
+++ b/test/pxp_lib_test/pxp_test.c
@@ -177,7 +177,7 @@ static void copy_image_to_fb(int left, int top, int width, int height, uint *img
 	bytes_per_pixel = screen_info->bits_per_pixel / 8;
 
 	for (i = 0; i < height; i++) {
-		memcpy(fb_ptr + ((i + top) * screen_info->xres + left) * bytes_per_pixel / 4,
+		memcpy(fb_ptr + ((i + top) * screen_info->xres_virtual + left) * bytes_per_pixel / 4,
 			img_ptr + (i * width) * bytes_per_pixel /4,
 			width * bytes_per_pixel);
 	}
@@ -195,6 +195,7 @@ static void update_to_display(int left, int top, int width, int height, int wave
 	upd_data.update_region.top = top;
 	upd_data.update_region.height = height;
 	upd_data.temp = TEMP_USE_AMBIENT;
+	upd_data.use_alt_buffer = 0;
 
 	if (wait_for_complete) {
 		/* Get unique marker value */
-- 
1.8.0

